mmdet/models/necks/bifpn.py

```diff
@@ -0,0 +1,222 @@
+                            size=(target_h, target_w),
+                            mode="nearest")
+                else:
+                    raise NotImplementedError
+                input_nodes.append(input_node)
+            # attention
+            weights = F.relu(
+                getattr(self, f"weights_f{feat_level}_{inputs_offsets_str}"))
+            norm_weights = weights / (weights.sum() + 0.0001)
+            new_node = torch.stack(input_nodes, dim=-1)
+            new_node = (norm_weights * new_node).sum(dim=-1)
+            new_node = new_node * new_node.sigmoid()
+            feats.append(
+                getattr(self,
+                        f"outputs_f{feat_level}"
+                        f"_{inputs_offsets_str}")(new_node))
+
+        output_feats = []
+        for idx in range(num_levels):
+            for i, fnode in enumerate(reversed(self.nodes)):
+                if fnode['feat_level'] == idx:
+                    output_feats.append(feats[-1 - i])
+                    break
+            else:
+                raise ValueError()
+        return output_feats
+
+
+@NECKS.register_module()
+class BiFPN(BaseModule):
+    """ The neck used in 'CenterNet2 <http://arxiv.org/abs/2103.07461>'_
+    Originally, the is was proposed  in
+    'EfficientDet https://arxiv.org/abs/1911.09070'_
+
+    Args:
+        in_channels (List[int]): Number of input channels per scale.
+        out_channels (int): Number of output channels (used at each scale)
+        num_outs (int): Number of output scales (implemented case 3 and 5).
+        start_level (int): Index of the start input backbone level used to
+            build the feature pyramid. Default: 0.
+        end_level (int): Index of the end input backbone level (inclusive) to
+            build the feature pyramid. Default: -1, which means the last level.
+        num_outs (int): Number of output features
+        conv_cfg (dict): Config dict for convolution layer. Default: None.
+        norm_cfg (dict): Config dict for normalization layer. Default: None.
+        act_cfg (dict): Config dict for activation layer in ConvModule.
+            Default: None.
+        init_cfg (dict or list[dict], optional): Initialization config dict.
+
+    """
+
+    def __init__(self,
+                 in_channels,
+                 out_channels,
+                 num_outs,
+                 start_level=0,
+                 end_level=-1,
+                 num_bifpn=4,
+                 conv_cfg=None,
+                 norm_cfg=dict(type='BN', requires_grad=True),
+                 act_cfg=None,
+                 init_cfg=dict(
+                     type='Normal',
+                     layer='Conv2d',
+                     std=0.01,)):
+        super(BiFPN, self).__init__(init_cfg)
+        assert isinstance(in_channels, list)
+        assert num_outs == 5 or num_outs == 3, NotImplementedError
+        self.num_bifpn = num_bifpn
+        self.__extra_levels = nn.ModuleList()
```

Comment:
The prefix  `—` seems unnecessary. In BiFPN, I believe following the name in FPN is a better choice.